src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -367,6 +367,13 @@ static String createQuery(Map<String, Object> params)
     } else if (value == null) {
       flatParams = new ArrayList<>();
       flatParams.add(new Parameter(keyPrefix, ""));
+    } else if ((value instanceof File) || (value instanceof InputStream)) {
+      throw new InvalidRequestException(String.format(
+          "java.io.File or java.io.InputStream %s is not supported at '%s' parameter. "
+          + "Please check our API reference for the parameter type, "
+          + "or use the provided parameter class instead.",
+          value, keyPrefix),
+          keyPrefix, null, null, 0, null);
```

Comment:
The error here could arise 
- if users pass file object to non-multipart request at any level.
- multi-part request params having nested file objects

Reply:
👍  Excellent, thanks for adding an explicit check and error.